src/game/bosses/boss_3.c

```diff
@@ -0,0 +1,1811 @@
+    gPlayer.moveState |= MOVESTATE_IGNORE_INPUT;
+
+    sub_8039ED4();
+
+    gPseudoRandom = gStageTime;
+
+    t = TaskCreate(Task_EggTotemMain, sizeof(EggTotem), 0x4000, 0,
+                   TaskDestructor_EggTotemMain);
+    totem = TASK_DATA(t);
+
+    if (DIFFICULTY_BOSS_IS_NOT_NORMAL) {
+        totem->lives = 6;
+    } else {
+        totem->lives = 8;
+    }
+
+    if (IS_FINAL_STAGE(gCurrentLevel)) {
+        totem->lives = totem->lives /= 2u;
+    }
+
+    if (IS_FINAL_STAGE(gCurrentLevel)) {
+        // _0803ED48
+        totem->qWorldX = Q(13714);
+        totem->qWorldY = Q(170);
+
+        for (i = 0; i < ARRAY_COUNT(totem->qWheelPos); i++) {
+            totem->qWheelPos[i].x = Q(13714);
+            totem->qWheelPos[i].y = Q(170);
+        }
+        // _0803EE06
+
+        for (i = 0; i < ARRAY_COUNT(totem->unk3C); i++) {
+            totem->unk3C[i].qWorldX = Q(13714);
+            totem->unk3C[i].qWorldY = Q(170) + sTotemDiscYs[i];
+
+            totem->unk3C[i].qUnk8 = 0;
+            totem->unk3C[i].qUnkA = 0;
+            totem->unk3C[i].qUnkC = SIN_DEG(270);
+            totem->unk3C[i].unk12 = 0;
+            totem->unk3C[i].unk13 = 0;
+            totem->unk3C[i].unkE = 0;
+            totem->unk3C[i].unk10 = 0;
+
+#ifndef NON_MATCHING
+            // This is inside a if(IS_FINAL_STAGE(gCurrentLevel)), anyway
+            if (IS_FINAL_STAGE(gCurrentLevel))
+#endif
+            {
+                totem->unk3C[i].unk14 = 1;
+            }
+
+            totem->unk3C[i].unk17 = 0;
+            totem->unk3C[i].unk15 = 0;
+            totem->unk3C[i].unk16 = 0;
+        }
+    } else {
+        // _0803EEB0
+        totem->qWorldX = Q(680);
+        totem->qWorldY = Q(170);
+
+        for (i = 0; i < ARRAY_COUNT(totem->qWheelPos); i++) {
+            totem->qWheelPos[i].x = Q(680);
+            totem->qWheelPos[i].y = Q(170);
+        }
+        // _0803EF72
+
+        for (i = 0; i < ARRAY_COUNT(totem->unk3C); i++) {
+            totem->unk3C[i].qWorldX = Q(680);
+            totem->unk3C[i].qWorldY = Q(170) + sTotemDiscYs[i];
+
+            totem->unk3C[i].qUnk8 = 0;
+            totem->unk3C[i].qUnkA = 0;
+            totem->unk3C[i].qUnkC = SIN_DEG(270);
+            totem->unk3C[i].unk12 = 0;
+            totem->unk3C[i].unk13 = 0;
+            totem->unk3C[i].unkE = 0;
+            totem->unk3C[i].unk10 = 0;
+
+            if (DIFFICULTY_BOSS_IS_NOT_NORMAL) {
+                totem->unk3C[i].unk14 = 1;
+            } else {
+                totem->unk3C[i].unk14 = 3;
+            }
+
+            totem->unk3C[i].unk17 = 0;
+            totem->unk3C[i].unk15 = 0;
+            totem->unk3C[i].unk16 = 0;
+        }
+    }
+    // _0803F038
+    totem->unk8 = Q(5);
+    totem->unkA = 0;
+    totem->unk33 = 240;
+    totem->unk30 = 0;
+    totem->unk34 = 0;
+    totem->unk35 = 0;
+    totem->unk36 = 0;
+
+    totem->unkB4 = 120;
+    totem->tilesBullet = VramMalloc((2 * 2) * 2);
+
+    s = &totem->sprC0.s;
```

Comment:
Do none of the init macros work for these?

Reply:
Or because it's non matching

Reply:
Mostly because it's nonmatching